src/openrct2/ride/Ride.cpp

```diff
@@ -4779,34 +4778,35 @@ static bool ride_create_vehicles(Ride* ride, const CoordsXYE& element, int32_t i
             }
         }
     }
-    ride_update_vehicle_colours(ride);
+    ride_update_vehicle_colours(this);
     return true;
 }
 
 /**
- *
+ * Move all the trains so each one will be placed at the block brake of a different block.
+ * The first vehicle will placed into the first block and all other vehicles in the blocks
+ * preceding that block.
  *  rct2: 0x006DDF9C
  */
-void loc_6DDF9C(Ride* ride, TileElement* tileElement)
+void Ride::MoveTrainsToBlockBrakes(const TileElement* firstBlock)
```

Comment:
You can simplify the logic slightly by passing in a TrackElement reference instead of the generic TileElement ptr.